### ISSUE TITLE
Unit Test cache is never found

### ISSUE THREAD

**Issue by sjmf** [DurhamARC/IASCPolls#2]
The test stage `cache/action@v3` in our unit-tests.yml Actions workflow never hits the cache. The `setup-miniconda` stage [supports caching](https://github.com/conda-incubator/setup-miniconda#caching) but something about how we're using it isn't saving the cache correctly. Possibly the wrong cache path is being used. To investigate.

**Comment by sjmf** [comment]
The post-cache action which runs at the end of the workflow throws this warning:

```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

TODO: Fix the path that the cache action uses

**Comment by sjmf** [comment]
Addressed in 5ec9391, but need to see if permissions error for restoring miniconda envs cache is hit again.

**Comment by sjmf** [comment]
Permissions error on restoring cache is still an issue:

```

Run actions/cache@v3
Received 0 of 780112635 (0.0%), 0.0 MBs/sec
Received 67108864 of 780112635 (8.6%), 32.0 MBs/sec
Received 134217728 of 780112635 (17.2%), 42.6 MBs/sec
Received 239075328 of 780112635 (30.6%), 57.0 MBs/sec
Received 297795584 of 780112635 (38.2%), 56.8 MBs/sec
Received 394264576 of 780112635 (50.5%), 62.6 MBs/sec
Received 436207616 of 780112635 (55.9%), 59.4 MBs/sec
Received 532676608 of 780112635 (68.3%), 63.5 MBs/sec
Received 599785472 of 780112635 (76.9%), 63.5 MBs/sec
Received 671088640 of 780112635 (86.0%), 63.9 MBs/sec
Received 754974720 of 780112635 (96.8%), 65.4 MBs/sec
Received 780112635 of 780112635 (100.0%), 64.7 MBs/sec
Cache Size: ~744 MB (780112635 B)
/usr/bin/tar -xf /home/runner/work/_temp/e167225e-254c-47ff-8dfe-1fcf5e3724f9/cache.tzst -P -C /home/runner/work/IASCPolls/IASCPolls --use-compress-program unzstd
/usr/bin/tar: ../../../../../usr/share/miniconda/envs: Cannot utime: Operation not permitted
/usr/bin/tar: ../../../../../usr/share/miniconda/envs: Cannot change mode to rwxr-xr-x: Operation not permitted
/usr/bin/tar: Exiting with failure status due to previous errors
Warning: Failed to restore: "/usr/bin/tar" failed with error: The process '/usr/bin/tar' failed with exit code 2
```
 See https://github.com/DurhamARC/IASCPolls/actions/runs/4960249906/jobs/8875487369

**Comment by sjmf** [comment]
Fix in 1ea0b37
